gtda/diagrams/features.py

```diff
@@ -388,3 +389,157 @@ def transform(self, X, y=None):
+        in a :obj:`joblib.parallel_backend` context. ``-1`` means using all
+        processors.
+
+    Attributes
+    ----------
+    homology_dimensions_ : list
+        Homology dimensions seen in :meth:`fit`, sorted in ascending order.
+
+    """
+    _hyperparameters = {
+        'polynomial_type': {'type': str,
+                            'in': _AVAILABLE_POLYNOMIALS.keys()},
+        'n_coefficients': {'type': (int, type(None)),
+                           'in': Interval(1, np.inf, closed='left')},
+    }
+
+    def __init__(self, polynomial_type='R', n_coefficients=10, n_jobs=None):
+        self.n_coefficients = n_coefficients
+        self.polynomial_type = polynomial_type
+        self.n_jobs = n_jobs
+
+    def fit(self, X, y=None):
+        """Store all observed homology dimensions in
+        :attr:`homology_dimensions_` and compute :attr:`n_coefficients_`. Then,
+        return the estimator.
+
+        This method is here to implement the usual scikit-learn API and hence
+        work in pipelines.
+
+        Parameters
+        ----------
+        X : ndarray of shape (n_samples, n_features, 3)
+            Input data. Array of persistence diagrams, each a collection of
+            triples [b, d, q] representing persistent topological features
+            through their birth (b), death (d) and homology dimension (q).
+            It is important that, for each possible homology dimension, the
+            number of triples for which q equals that homology dimension is
+            constants across the entries of `X`.
+
+        y : None
+            There is no need for a target in a transformer, yet the pipeline
+            API requires this parameter.
+
+        Returns
+        -------
+        self : object
+
+        """
+        validate_params(
+            self.get_params(), self._hyperparameters, exclude=['n_jobs'])
+        X = check_diagrams(X)
+
+        # Find the unique homology dimensions in the 3D array X passed to `fit`
+        # assuming that they can all be found in its zero-th entry
+        self.homology_dimensions_ = sorted(set(X[0, :, 2]))
+
+        if self.n_coefficients is None:
+            self.n_coefficients_ = \
+                np.array([X[i].shape[0] for i in range(len(X))]).max()
+        else:
+            self.n_coefficients_ = self.n_coefficients
+
+        self._polynomial_function = \
+            _implemented_polynomial_recipes[self.polynomial_type]
+
+        return self
+
+    def _complex_polynomial(self, X):
+        Xt = np.zeros(2 * self.n_coefficients_, )
+        X = X[X[:, 0] != X[:, 1]]
+
+        roots = self._polynomial_function(X)
+        coefficients = np.poly(roots)
+
+        coefficients = np.array(coefficients[::-1])[1:]
+        dimension = min(self.n_coefficients_, coefficients.shape[0])
+        Xt[:dimension] = coefficients[:dimension].real
+        Xt[self.n_coefficients_:self.n_coefficients_ + dimension] = \
+            coefficients[:dimension].imag
```

Comment:
It seems to me that with this implementation, there is a slight mismatch with the paper:
Consider the polynomials `ax^2 + bx + c` and `ex + f` with `n_coefficients = 2`,
Expected: 
`[Re(b), Re(a), Im(b), Im(a)]` and `[0, Re(e), 0, Im(e)]`

Actual:
`[Re(b), Re(a), Im(b), Im(a)]` and `[Re(e), 0, Im(e), 0]`

Reading the paper:
> We manage this problem by adding n−m null coefficients to f_D′(t), i.e. multiplying f_D′(t) by t^(n−m)

Hence, due to the fact that we invert coefficients in line 502, i believe we would need to align the last non-zero entries corresponding to real parts.

Reply:
Alternatively, we could modify line 502 to 
`coefficients = np.array(coefficients[::-1])[1:]` ?

Reply:
I will check! As for your solution, it seems like there is no change to L502 in it :)

Reply:
You're right: I meant
`coefficients = np.array(coefficients)[:-1]`

Reply:
Oh, yes, makes perfect sense.